src/Components/Web.Extensions/src/InputFile/IFileListEntry.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Represents the data of a file selected from an <see cref="InputFile"/> component.
+    /// </summary>
+    public interface IFileListEntry
+    {
+        /// <summary>
+        /// Gets the name of the file.
+        /// </summary>
+        string? Name { get; }
```

Comment:
Just curious: is it really possible for this to be null? Are there known scenarios where a browser might choose not to supply any name?

Reply:
Looks like you're right - worst case, browsers are required to supply the empty string: https://w3c.github.io/FileAPI/#file-attrs.